thunder/core/jit_ext.py

```diff
@@ -1347,6 +1347,19 @@ def bind_inputs(name, trace, input_vars, input_proxies):
     trace.args = input_proxies
 
 
+def _get_process_group_from(*fn_and_args) -> Optional["ProcessGroup"]:
+    # `ddp` and `fsdp` transforms add attribute `procses_group_for_ddp`
+    # on the Module that they wrap. This module could be passed to `thunder.jit`
+    # as the function to be jitted or as an argument of the function to be jitted.
+    # NOTE: `ddp` and `fsdp` use default process group. So finding the first
+    #       occurence is sufficient.
+    for fn_or_arg in fn_and_args:
+        if (pg := getattr(fn_or_arg, "process_group_for_ddp", None)) is not None:
```

Comment:
Nit: We're not starved for vertical space, so as a rule of thumb I would not use := if the number of non-whitespace characters get smaller from it.

Reply:
More broadly, I think we would have to check if it's all the same process group rather than assuming it.

Are we making fsdp/ddp "spill" over from one model to more here? Is that working as intended?